multimedia/VLC2/Portfile

```diff
@@ -299,8 +307,8 @@ if {(${subport} eq ${name}) || (${subport} eq "lib${name}")} {
         configure.args-replace  --disable-shout --enable-shout
     }
 
-    variant smb description {Enable Samba 3 input support} {
-        depends_lib-append      port:samba3
+    variant smb description {Enable Samba 3+ input support} {
+        depends_lib-append      path:${prefix}/bin/smbclient:samba3
         configure.args-replace  --disable-smbclient --enable-smbclient
         configure.ldflags-append    -L${prefix}/lib/samba3
```

Comment:
It's redundant to put `${prefix}/` into that path depspec.

But look below that line where it says `-L${prefix}/lib/samba3`. That won't work for your proposed samba4 port, will it? If this ends up linking with a samba library, and samba3 and samba4 install them to different locations or names as I hope they do so that they don't conflict, then I assume there will need to be separate samba3 and samba4 variants.

Reply:
Shoot, I knew that of course. Your other concern is unnecessary. The samba4 port conflicts with samba3, installs its libraries into $prefix/lib/samba and a few additional symlinks in $prefix/lib/samba3 (e.g. for dependents that don't use the pkgconfig files). That directory would be temporary unless there is a need to preserve two samba ports (I haven't run into any ABI compatibility problems IIRC).

I should get back to that project...

Reply:
The current samba4 port does not declare itself to be conflicting with samba3. But it was last updated over a decade ago and [has not been buildable](https://trac.macports.org/ticket/38639) for over 7 years at least, so we can basically consider the port not to exist at all; it's been dead and buried for a long time. Now if you're planning to resurrect samba4 by updating it to a current version that's great, I'm just asking you to ensure that it does not conflict with samba3. It would be unusual for a port named samba4 to install any files in a directory named samba3. If you don't see a need for separate samba3 and samba4 ports, then by all means have just one. At that point, since we would be changing the name away from samba3 regardless, it might make sense to call the new port just "samba". And then there can indeed be just a single samba or smb variant for vlc.

Reply:
I don't disagree with all that; this change is indeed a temporary compromise that I should probably have kept private. If ever I do get the samba4 port to work sufficiently well all dependents will surely require at least a revbump.

FWIW, I've delved into samba4 because apparently the port:samba3 nor the system smb version can connect to the Win10 computers in the house. I've managed to get v4.8.9 to build following online instructions, and now I can again connect to those computers. Not via the Finder, of course, and only most of the time (sometimes it still just doesn't work). The online instructions also outline how to set up a server, and I haven't yet managed to get that to function. The feature doesn't interest me myself, but others might expect to have it...